### PR TITLE
Benchmarks in csv format

### PR DESCRIPTION
**Pull request by outscale-jju**

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -402,6 +402,23 @@ bench: bench-antispoof bench-core bench-diode bench-firewall bench-nic bench-pri
 	$(srcdir)/tests/pmtud/bench.sh
 	$(srcdir)/tests/tap/bench.sh
 	$(srcdir)/tests/ip-fragment/bench.sh
+
+benchmark.%: bench-antispoof bench-core bench-diode bench-firewall bench-nic bench-print bench-queue bench-switch bench-vhost bench-pmtud bench-vtep bench-tap bench-rxtx bench-ip-fragment
```
Comment (outscale-mgo) [inline Makefile.am]: maybe a tmp variable for the bench list would be useful here

File: tests/core/bench-nop.c [hunk 2]
```diff
@@ -42,7 +42,13 @@ void test_benchmark_nop(void)
 	struct ether_addr mac2 = {{0x52,0x54,0x00,0x12,0x34,0x21}};
 	uint32_t len;
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "nop", argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/core/bench-nop.c]: or you can simply `g_assert(!error)` as we do everywhere else

File: tests/diode/bench-diode.c [hunk 3]
```diff
@@ -42,7 +42,13 @@ void test_benchmark_diode(void)
 	struct ether_addr mac2 = {{0x52,0x54,0x00,0x12,0x34,0x21}};
 	uint32_t len;
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "diode", argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/diode/bench-diode.c]: g_assert(!error)

File: tests/firewall/bench-firewall.c [hunk 4]
```diff
@@ -44,7 +44,13 @@ void test_benchmark_firewall(void)
 	uint32_t ip_dst;
 	uint32_t len;
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "firewall", argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/firewall/bench-firewall.c]: g_assert(!error)

File: tests/ip-fragment/bench.c [hunk 5]
```diff
@@ -44,7 +46,14 @@ static void test_benchmark_ip_fragment(int mtu, int max_burst_cnt)
 	struct pg_bench_stats stats;
 	int pkt_len = 1500;
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, title, argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/ip-fragment/bench.c]: g_assert(!error)

File: tests/ip-fragment/bench.c [hunk 6]
```diff
@@ -98,7 +108,13 @@ static void test_benchmark_ip_defragment(int mtu)
 	struct rte_mbuf **tmp_pkts;
 	uint64_t tmp_mask = 1;
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, title, argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/ip-fragment/bench.c]: g_assert(!error)

File: tests/ip-fragment/bench.c [hunk 7]
```diff
@@ -162,7 +179,13 @@ static void benchmark_ip_frag_and_defragment(int mtu)
 	struct rte_mbuf **tmp_pkts;
 	uint64_t tmp_mask = pg_mask_firsts(64);
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, title, argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/ip-fragment/bench.c]: g_assert(!error)

File: tests/nic/bench-nic.c [hunk 8]
```diff
@@ -67,7 +67,13 @@ void test_benchmark_nic(void)
 		}
 	}
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "nic", argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/nic/bench-nic.c]: g_assert(!error)

File: tests/pmtud/bench.c [hunk 9]
```diff
@@ -35,7 +35,13 @@ static void test_benchmark_pmtud(void)
 	uint32_t len;
 	struct pg_bench_stats stats;
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "pmtud", argc, argv, &error) < 0) {
```
Comment (outscale-mgo) [inline tests/pmtud/bench.c]: g_assert(!error)

File: tests/print/bench-print.c [hunk 10]
```diff
@@ -42,7 +42,13 @@ void test_benchmark_print(void)
 	struct ether_addr mac2 = {{0x52,0x54,0x00,0x12,0x34,0x21}};
 	uint32_t len;
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "print", argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/print/bench-print.c]: g_assert(!error)

File: tests/queue/bench-queue.c [hunk 11]
```diff
@@ -62,7 +62,13 @@ void test_benchmark_queue(void)
 		g_assert(0);
 	}
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "queue", argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/queue/bench-queue.c]: g_assert(!error)

File: tests/rxtx/bench-rxtx.c [hunk 12]
```diff
@@ -94,7 +94,13 @@ void test_benchmark_rxtx(void)
 	rxtx_enter = pg_rxtx_new("enter", &myrx, NULL, pd);
 	rxtx_exit = pg_rxtx_new("exit", NULL, &mytx, pd);
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "rxtx", argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/rxtx/bench-rxtx.c]: g_assert(!error)

File: tests/switch/bench-switch.c [hunk 13]
```diff
@@ -44,7 +44,13 @@ void test_benchmark_switch(void)
 
 	sw = pg_switch_new("switch", 20, 20, DEFAULT_SIDE, &error);
 	g_assert(!error);
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "switch", argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/switch/bench-switch.c]: g_assert(!error)

File: tests/tap/bench-tap.c [hunk 14]
```diff
@@ -68,7 +68,13 @@ void test_benchmark_tap(void)
 	run_ok("ip netns exec bench brctl addif br0 bench1");
 	run_ok("ip netns exec bench ip link set br0 up");
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "tap", argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/tap/bench-tap.c]: g_assert(!error)

File: tests/vhost/bench-vhost.c [hunk 15]
```diff
@@ -55,7 +59,13 @@ void test_benchmark_vhost(char *vm_image_path,
 	int ret;
 	int qemu_pid;
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "vhost", argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/vhost/bench-vhost.c]: g_assert(!error)

File: tests/vtep/bench-vtep.c [hunk 16]
```diff
@@ -78,7 +78,14 @@ static void inside_to_vxlan(void)
 	struct ether_addr mac3 = {{0x52,0x54,0x00,0x12,0x34,0x31}};
 	uint32_t len;
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, "vtep inside to vxlan",
+			  argc, argv, &error) < 1) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/vtep/bench-vtep.c]: g_assert(!error)

File: tests/vtep/bench-vtep.c [hunk 17]
```diff
@@ -162,7 +168,13 @@ static void vxlan_to_inside(int flags)
 			   mac_vtep, PG_VTEP_DST_PORT, flags, &error);
 	g_assert(!error);
 
-	pg_bench_init(&bench);
+	if (pg_bench_init(&bench, title, argc, argv, &error) < 0) {
+		if (pg_error_is_set(&error)) {
```
Comment (outscale-mgo) [inline tests/vtep/bench-vtep.c]: g_assert(!error)

**Comment by outscale-jju** [comment]
merged

**Pull request closed without merging**